### DISCUSSION TITLE
Access to User > Preferences without write access

### DISCUSSION THREAD

**Discussion by HeinzWuert** [kimai/kimai#4335] [Questions & Answers]
The preferences should also be visible to team leaders.
Unfortunately, these are not write-protected. Is there a way to adapt this? Now there is no option to differentiate between reading and writing under roles; you can only control global access.

**Comment by kevinpapst** [comment]
They should be visible in the user listing, if you can activate that column.

**Comment by HeinzWuert** [comment]
Roles > preferences_other_profile 
includes write access. Thats the problem.

**Reply by kevinpapst** [reply]
That is not my point. 
I wanted to give you are workaround.
If a teamlead wants to see the roles, that is already possible in the "users" screen, you just have to show the "Roles" column.

**Comment by HeinzWuert** [comment]
Teamleads should see the preferences with the addon custom fields.
This only can be enabled via preferences_other_profile including write access
A solution would be two roles:
- view preferences_other_profile
- edit preferences_other_profile

**Reply by kevinpapst** [reply]
Oh wow, we are talking about different things.

> differentiate between reading and writing under roles

I thought you talk about user roles: that someone should be able to see the Roles tab only for reading.

You can open a feature request. But read access to user preferences is not on my roadmap. I can't remember that anyone ever asked for it.

**Comment by HeinzWuert** [comment]
would be nice when using custom fields as we do

**Comment by lucas-0liveira** [comment]
Thanks for bringing this up. It's crucial for team leaders to have visibility into preferences. It's concerning to hear that they're not write-protected, as this could potentially lead to unintended changes. Is there a way we can work on adapting this feature? Currently, it seems like there's no option to differentiate between reading and writing under roles, which limits our control over access. Let's explore potential solutions to ensure preferences are both accessible and secure for our team leaders.

**Reply by kevinpapst** [reply]
Stop posting ChatGPT generated non-sense or I will block you.
